ocaml/xapi/repository.ml

```diff
@@ -360,34 +368,29 @@ let parse_updateinfo ~__context ~self =
 let get_pool_updates_in_json ~__context ~hosts =
   try
     let repository = get_single_enabled_update_repository ~__context in
-    let funs =
-      List.map (fun host ->
-          fun () ->
-            ( (Ref.string_of host),
-              (http_get_host_updates_in_json ~__context ~host ~installed:true) )
-        ) hosts
-    in
-    let rets =
-      with_pool_repositories (fun () ->
-          Helpers.run_in_parallel funs capacity_in_parallel)
-    in
-    let repository_name = get_repository_name  ~__context ~self:repository in
-    let updates_info = parse_updateinfo ~__context ~self:repository in
-    let updates_of_hosts, ids_of_updates =
-      rets
-      |> List.fold_left (fun (acc1, acc2) (host, ret_of_host) ->
-          let json_of_host, uids =
-            consolidate_updates_of_host ~repository_name ~updates_info host ret_of_host
-          in
-          ( (json_of_host :: acc1), (UpdateIdSet.union uids acc2) )
-        ) ([], UpdateIdSet.empty)
-    in
-    `Assoc [
-      ("hosts", `List updates_of_hosts);
-      ("updates", `List (UpdateIdSet.elements ids_of_updates
-                         |> List.map (fun uid  ->
-                             UpdateInfo.to_json (List.assoc uid updates_info))));
-      ("hash", `String (Db.Repository.get_hash ~__context ~self:repository))]
+    (match !updates_in_cache with
+     | None ->
+         raise Api_errors.(Server_error (updates_require_sync, []))
```

Comment:
Is there a reason not to do the sync here? Then we don't need a new API error

Reply:
`sync` is a API call. I don't think this should be done in HTTP endpoint `/updates` which is kind of another API. In other words, this would mean a API call contains another API call.